addons/website_slides_survey/tests/test_course_certification_failure.py

```diff
@@ -98,11 +101,14 @@ def test_course_certification_failure_flow(self):
 
         self.slide_certification.with_user(self.user_portal)._generate_certification_url()
         self.assertTrue(channel_partner.active, 'Portal user membership should be a unarchived upon joining the course once again')
-        self.assertEqual(1, len(slide_partner.user_input_ids.filtered(lambda user_input: user_input.state != 'done')), 'A new user input should have been automatically created upon slide view')
+        self.assertEqual(1, len(slide_partner.user_input_ids.filtered(lambda user_input: user_input.slide_id)), 'A new user input should have been automatically created upon slide view')
 
         # Step 8: fill in the created user_input with correct answers this time
-        self.fill_in_answer(slide_partner.user_input_ids.filtered(lambda user_input: user_input.state != 'done'), certification.question_ids, good_answers=True)
+        self.fill_in_answer(slide_partner.user_input_ids.filtered(
+            lambda user_input: user_input.slide_id), certification.question_ids, good_answers=True)
```

Comment:
Unless it is a more recent feature, this is shorter:
 ```suggestion
        self.fill_in_answer(slide_partner.user_input_ids.filtered('slide_id'), certification.question_ids, good_answers=True)
```